src/main/java/io/gravitee/policy/http/redirect/HTTPRedirectPolicy.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.http.redirect;
+
+import static java.util.stream.Collectors.toMap;
+
+import io.gravitee.el.TemplateEngine;
+import io.gravitee.gateway.api.ExecutionContext;
+import io.gravitee.gateway.api.Request;
+import io.gravitee.gateway.api.Response;
+import io.gravitee.gateway.api.http.HttpHeaderNames;
+import io.gravitee.gateway.reactive.api.context.http.HttpPlainExecutionContext;
+import io.gravitee.gateway.reactive.api.policy.http.HttpPolicy;
+import io.gravitee.policy.api.PolicyChain;
+import io.gravitee.policy.api.PolicyResult;
+import io.gravitee.policy.api.annotations.OnRequest;
+import io.gravitee.policy.http.redirect.configuration.HTTPRedirectPolicyConfiguration;
+import io.gravitee.policy.http.redirect.configuration.HTTPRedirectPolicyConfiguration.Rule;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Maybe;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class HTTPRedirectPolicy implements HttpPolicy {
+
+    private static final String GROUP_TEMPLATE_ATTRIBUTE = "group";
+    private static final String GROUP_NAME_ATTRIBUTE = "groupName";
+    private static final String INTERRUPT_KEY = "REQUEST_REDIRECTED";
+
+    private final HTTPRedirectPolicyConfiguration configuration;
+
+    @Override
+    public String id() {
+        return "policy-http-redirect";
+    }
+
+    @Override
+    public Completable onRequest(final HttpPlainExecutionContext ctx) {
+        return rxFindMatchingRule(ctx.request().pathInfo(), ctx.getTemplateEngine())
+            .flatMap(ruleMatch -> rxBuildRedirect(ruleMatch, ctx.getTemplateEngine()))
+            .flatMapCompletable(redirect -> rxSendRedirect(redirect, ctx));
+    }
+
+    private Maybe<RuleMatch> rxFindMatchingRule(String path, TemplateEngine templateEngine) {
+        return Flowable
+            .fromIterable(configuration.rules())
+            .flatMapMaybe(rule ->
+                templateEngine
+                    .eval(rule.path(), String.class)
+                    .map(Pattern::compile)
```

Comment:
I would avoid compiling pattern on each request call.
As the policy configuration is cached, better to initialize the patterns once, so you can reuse them.

Reply:
@brasseld but I just figured out that the compile applies on a previously compiled EL, which makes the cache key dynamic, would that work ?

Reply:
ah, indeed, I have not noticed that the pattern would be dynamic. 
I'm sure we can still find a way to cache the PAttern to avoid recompilation, even if the pattern is dyanmic.
pattern compilation is quite costly fyi

Reply:
@brasseld would that be ok to cache on first request ? if people don't use EL, only that first request would carry the cost of pattern compilation

Reply:
@brasseld done